### PR TITLE
Drop support for Python2 and Django < 1.10

### PR DESCRIPTION
**Pull request by hugorodgerbrown**
Provides dual-support for Django 1.11 and 2.0, drops support for python2.

* Convert `is_authenticated(), is_anonymous()` to properties
* Add `on_delete` to all foreign keys, and update migrations
* Update `urls.py` to handle 1.11 and 2.0 `re_path` format
* Update `RequestTokenMiddleware` to drop use of `MiddlewareMixin` in favour of new style
* Remove use of `six`

### CODE REVIEW

File: request_token/models.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def validate_max_uses(self):
 
     def _auth_is_anonymous(self, request):
         """Authenticate anonymous requests."""
-        assert request.user.is_anonymous(), 'User is authenticated.'
+        assert request.user.is_anonymous, 'User is authenticated.'
```
Comment (djm) [inline request_token/models.py]: Ah, this is bad. `assert` can be stripped by the compiler so this code could effectively be a `pass` on those that do so. We don't run with compiler optimisations stripping the asserts, but as this is OS I feel we have a little duty to fix this up.

Can be replaced with a conditional instead, which never get stripped.

More: https://dbader.org/blog/python-assert-tutorial

File: request_token/models.py [hunk 2]
```diff
@@ -265,7 +265,7 @@ def _auth_is_anonymous(self, request):
 
     def _auth_is_authenticated(self, request):
         """Authenticate requests with existing users."""
-        assert request.user.is_authenticated(), 'User is anonymous.'
+        assert request.user.is_authenticated, 'User is anonymous.'
```
Comment (djm) [inline request_token/models.py]: Ditto ^

File: request_token/models.py (hunk 1 above)
Comment (hugorodgerbrown) [reply request_token/models.py]: True - but not part of this PR.

**Comment by hugorodgerbrown** [comment]
@djm asserts removed

**Pull request merged**